prime-router/src/main/kotlin/azure/SendFunction.kt

```diff
@@ -104,13 +103,15 @@ class SendFunction(
                 val nextRetryItems = mutableListOf<String>()
                 val externalFileName = Report.formExternalFilename(header, workflowEngine.reportService)
                 val sentReportId = UUID.randomUUID() // each sent report gets its own UUID
-                if (receiver.transport == null) {
+                val retryItems = retryToken?.items
+                val transport = getTransport(receiver.transportType)
+                if (transport == null) {
                     actionHistory.setActionType(TaskAction.send_warning)
-                    actionHistory.trackActionResult("Not sending $inputReportId to $serviceName: No transports defined")
+                    actionHistory
+                        .trackActionResult("Not sending $inputReportId to $serviceName: Unsupported transport type")
```

Comment:
Left it in because I'd wanted to make sure we had some kind of alert in the case of someone adding a transport type, but not adding a corresponding transport for it. I didn't know I could make it just not compile with a sealed class, so changed to that per your suggestion.